File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/wallet/ImportWalletTest.java
package org.hyperledger.indy.sdk.wallet;

import org.hyperledger.indy.sdk.IOException;
import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.did.Did;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class ImportWalletTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testImportWalletWorks() throws Exception {
		String did = Did.createAndStoreMyDid(wallet, "{}").get().getDid();
		Did.setDidMetadata(wallet, did, METADATA).get();

		String didWithMetaBefore = Did.getDidWithMeta(wallet, did).get();

		Wallet.exportWallet(wallet, EXPORT_CONFIG_JSON).get();

		wallet.closeWallet().get();
		Wallet.deleteWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();

		Wallet.importWallet(WALLET_CONFIG, WALLET_CREDENTIALS, EXPORT_CONFIG_JSON).get();

		wallet = Wallet.openWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();

		String didWithMetaAfter = Did.getDidWithMeta(wallet, did).get();

		assertEquals(didWithMetaBefore, didWithMetaAfter);
	}

	@Test
	public void testImportWalletWorksForNotExists() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(IOException.class));

		Wallet.importWallet(WALLET_CONFIG, WALLET_CREDENTIALS, EXPORT_CONFIG_JSON).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/WalletTests/ImportWalletTest.cs
using Hyperledger.Indy.DidApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.WalletTests
{
    [TestClass]
    public class ImportWalletTest : IndyIntegrationTestWithSingleWallet
    {
        [TestMethod]
        public async Task TestImportWalletWorks()
        {
            var result = await Did.CreateAndStoreMyDidAsync(wallet, "{}");
            var did = result.Did;

            await Did.SetDidMetadataAsync(wallet, did, METADATA);

            var didWithMetaBefore = await Did.GetDidMetadataAsync(wallet, did);

            await wallet.ExportAsync(EXPORT_CONFIG_JSON);

            await wallet.CloseAsync();
            await Wallet.DeleteWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            await Wallet.ImportAsync(WALLET_CONFIG, WALLET_CREDENTIALS, EXPORT_CONFIG_JSON);

            wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            var didWithMetaAfter = await Did.GetDidMetadataAsync(wallet, did);

            Assert.AreEqual(didWithMetaBefore, didWithMetaAfter);
        }

        [TestMethod]
        public async Task TestImportWalletWorksForNotExists()
        {
            var ex = await Assert.ThrowsExceptionAsync<IOException>(() =>
                Wallet.ImportAsync(WALLET_CONFIG, WALLET_CREDENTIALS, EXPORT_CONFIG_JSON)
            );
        }
    }
}
